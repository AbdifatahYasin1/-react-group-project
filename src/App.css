reset

* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

.navigation {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.navigation ul {
  display: flex;
  list-style: none;
}

.navigation ul li a {
  margin: 0 10px;
  text-decoration: none;
  color: black;
  font-size: 20px;
}

.navigation ul li a:hover {
  color: #2b2828;
}

.logo-container {
  display: flex;
  align-items: center;
}

.logo {
  width: 100px;
  height: 100px;
  margin: 0 10px;
}
.active{

  border-bottom: 2px solid #2b2828;
}
.inactive
{
  border-bottom: 2px solid transparent;
}