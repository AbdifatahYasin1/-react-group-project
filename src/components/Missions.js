import React, { useEffect, useRef } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Table } from 'react-bootstrap';
import { fetchMissions, joinMission, leaveMission } from '../redux/missions/missions';
import 'bootstrap/dist/css/bootstrap.min.css';

const Missions = () => {
  const dispatch = useDispatch();
  const missions = useSelector((state) => state.missions);
  const missionsRef = useRef(null);

  useEffect(() => {
    dispatch(fetchMissions());
  }, []);

  const handleJoin = (mission) => {
    dispatch(joinMission(mission));
  };

  const handleLeave = (mission) => {
    dispatch(leaveMission(mission));
  };

  return (
    <div className="missions-container">
      <Table striped bordered hover ref={missionsRef}>
        <thead>
          <tr>
            <th>Mission</th>
            <th>Description</th>
            <th>Status</th>
            <th> </th>
          </tr>
        </thead>
        <tbody>
          {
           missions.map((mission) => (
             <tr key={mission.mission_id}>
               <td className="col-2">{mission.mission_name}</td>
               <td className="col-6">{mission.description}</td>
               <td className="col-2 text-center pt-5">
                 {mission.reserved ? (<span className="bg-info text-white rounded-1 p-1">Active Member</span>)
                   : (<span className="bg-secondary text-white rounded-1 p-1">NOT A MEMBER</span>)}
               </td>
               <td className="col-2 text-center pt-5">
                 {!mission.reserved && (
                 <button
                   type="submit"
                   className="btn btn-light border-2 border-secondary"
                   onClick={() => dispatch(handleJoin(mission))}
                 >
                   Join Mission

                 </button>
                 )}
                 {mission.reserved && (
                 <button
                   name="leaveMission"
                   type="submit"
                   className="btn btn-light border-2 border-danger"
                   onClick={() => dispatch(handleLeave(mission))}
                 >
                   Leave Mission

                 </button>
                 )}
               </td>
             </tr>
           ))
}
        </tbody>
      </Table>
    </div>
  );
};

export default Missions;
