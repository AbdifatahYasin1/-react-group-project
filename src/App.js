import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import './App.css';
import Header from './components/Header';
import Missions from './components/Missions';
import Rockets from './components/Rockets';
import MyProfile from './components/Myprofile';

function App() {
  return (
    <Router>
      <Header />
      <Routes>
        <Route path="/" element={<Rockets />} />
        <Route path="/missions" element={<Missions />} />
        <Route path="/myprofile" element={<MyProfile />} />
      </Routes>
    </Router>
  );
}

export default App;
