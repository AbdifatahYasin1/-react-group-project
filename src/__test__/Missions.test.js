/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/store';
import Missions from '../components/Missions';
import App from '../App';

describe('Missions', () => {
  it('renders Missions component', () => {
    const m = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Missions />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(m).toMatchSnapshot();
  });
});

const MissionMock = () => (
  <Provider store={store}>
    <App />
  </Provider>
);







