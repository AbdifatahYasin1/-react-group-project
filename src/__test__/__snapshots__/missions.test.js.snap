// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions renders Missions component 1`] = `
<div
  className="missions-container"
>
  <table
    className="table table-striped table-bordered table-hover"
  >
    <thead>
      <tr>
        <th>
          Mission
        </th>
        <th>
          Description
        </th>
        <th>
          Status
        </th>
        <th>
           
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
